app/models/book.rb

```diff
@@ -5,6 +5,8 @@ class Book < ApplicationRecord
   has_many :categories, through: :book_categories
   has_many :book_tags, inverse_of: :book, dependent: :destroy
   has_many :tags, through: :book_tags
+  has_many :orders, dependent: :destroy
+  has_many :carts, through: :orders
```

Comment:
Do we need this relationship?

Reply:
I think `has_many :orders, dependent: :destroy` is necessary because when you deleted a book, the record related to this book in the orders table should be deleted too.

`has_many :carts, through: :orders` - I will remove it.